### PR TITLE
#1772 Replace React-hot-loader with React-fast-refresh

### PR DESCRIPTION
**Pull request by srikant-ch5**
Fixes: #1772 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by srikant-ch5** [comment]
Reference: https://github.com/gaearon/react-hot-loader/issues/1227#issuecomment-482518698

**Comment by nmgokhale** [comment]
This change works fine when I have `checkbox_paramDef.json` open, and I make a change in `.jsx or .scss` file, it gets reflected on the UI without error. 


But when I make a change in `.jsx or .scss` file first and then I try to open `checkbox_paramDef.json` file, it throws a lot of errors.

**Comment by srikant-ch5** [comment]
> This change works fine when I have `checkbox_paramDef.json` open, and I make a change in `.jsx or .scss` file, it gets reflected on the UI without error.
> 
> But when I make a change in `.jsx or .scss` file first and then I try to open `checkbox_paramDef.json` file, it throws a lot of errors.

Hi @nmgokhale could you please verify latest changes to this PR in your local. I have replaced react-hot-loader with react-fast-refresh package.

File: canvas_modules/harness/package.json [hunk 1]
```diff
@@ -73,7 +74,7 @@
     "react-inlinesvg": "4.1.3",
     "react-intl": "6.6.3",
     "react-redux": "7.2.8",
-    "react-refresh": "0.14.0",
+    "react-refresh": "^0.14.0",
```
Comment (matthoward366) [inline canvas_modules/harness/package.json:77]: Any reason you're adding ^ to the versions?  I think most if not all these should be static versions.

Comment (srikant-ch5) [reply canvas_modules/harness/package.json]: Hi @matthoward366 It got added when I ran `npm install -D @pmmmwh/react-refresh-webpack-plugin react-refresh` now I have reverted it.

File: canvas_modules/harness/package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "@carbon/charts-react": "1.15.3",
     "@carbon/react": "^1.52.0",
     "@elyra/canvas": "file:../common-canvas",
+    "@pmmmwh/react-refresh-webpack-plugin": "^0.5.11",
```
Comment (nmgokhale) [inline canvas_modules/harness/package.json:41]: Remove `react-hot-loader` from package.json and pin this dependency as well.

Comment (srikant-ch5) [reply canvas_modules/harness/package.json]: Hi @nmgokhale I have removed react-hot-loader package and pinned react-refresh-webpack-plugin in package.json, please review.

**Review by nmgokhale — APPROVED** [review]

**Review by matthoward366 — APPROVED** [review]

**Pull request merged**